examples/regression/2-advanced-analysis/plot_coverage_validity.py

```diff
@@ -0,0 +1,528 @@
+plt.plot(cumulative_averages_mapie, alpha=0.5, label='MAPIE', color='g')
+
+plt.hlines(exact_cov, 0, num_splits, color='r', ls='--', label='Exact Cov.')
+plt.hlines(lower_bound, 0, num_splits, color='k', label='Lower Bound')
+plt.hlines(upper_bound, 0, num_splits, color='b', label='Upper Bound')
+
+plt.xlabel(r'Split Number')
+plt.ylabel(r'$\overline{\mathbb{C}}$')
+plt.title(r'$|D_{cal}| = $' + str(n_calib) + r' and $\delta = $' + str(delta))
+
+plt.legend(loc="upper right", ncol=2)
+plt.ylim(0.7, 1)
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# It can be seen that the two curves overlap, proving that both methods
+# produce the same results. Their effective coverage stabilizes between
+# the theoretical limits, always above the target coverage and converges
+# towards the exact coverage (i.e. expected according to the theory).
+
+
+##############################################################################
+# Experiment 2: Again but without fixing random_state
+# ---------------------------------------------------
+#
+# We just propose to reproduce the previous experiment without fixing the
+# random_state. The methods therefore follow different trajectories but
+# always achieve the expected coverage.
+
+# Run the experiment
+empirical_coverages_ref = []
+empirical_coverages_mapie = []
+
+for i in range(1, num_splits):
+    # Compute empirical coverage for each trial with StandardConformalizer
+    conformalizer = StandardConformalizer(model, non_conformity_func, delta)
+    coverage = get_coverage_prefit(
+        conformalizer, X_cal_test, y_cal_test, delta, n_calib
+    )
+    empirical_coverages_ref.append(coverage)
+
+    # Compute empirical coverage for each trial with MapieRegressor
+    conformalizer = MapieRegressor(estimator=model, cv="prefit")
+    coverage = get_coverage_prefit(
+        conformalizer, X_cal_test, y_cal_test, delta, n_calib
+    )
+    empirical_coverages_mapie.append(coverage)
+
+cumulative_averages_ref = cumulative_average(empirical_coverages_ref)
+cumulative_averages_mapie = cumulative_average(empirical_coverages_mapie)
+
+# Plot the results
+fig, ax = plt.subplots()
+plt.plot(cumulative_averages_ref, alpha=0.5, label='SplitCP', color='r')
+plt.plot(cumulative_averages_mapie, alpha=0.5, label='MAPIE', color='g')
+
+plt.hlines(exact_cov, 0, num_splits, color='r', ls='--', label='Exact Cov.')
+plt.hlines(lower_bound, 0, num_splits, color='k', label='Lower Bound')
+plt.hlines(upper_bound, 0, num_splits, color='b', label='Upper Bound')
+
+plt.xlabel(r'Split Number')
+plt.ylabel(r'$\overline{\mathbb{C}}$')
+plt.title(r'$|D_{cal}| = $' + str(n_calib) + r' and $\delta = $' + str(delta))
+
+plt.legend(loc="upper right", ncol=2)
+plt.ylim(0.7, 1)
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# Section 2: Comparison with different MAPIE CP methods
```

Comment:
Not sure that this example adds a lot of insights. I would make this example shorter by getting rid of the `prefit_asym` part.

Reply:
In practice, this part could be improved by adding more MAPIE methods (like split, CV+, CQR etc). In any case, we let the user test it themselves.